### PR TITLE
OCPBUILD-9: Update openshift/api and vendored CRDs

### PR DESCRIPTION
**Pull request by apoorvajagtap**
With [openshift/api/pull/1814/files](https://github.com/openshift/api/pull/1814/files#diff-e44e79a4fc37bb9c56990e53e03326faceb391db047ef1e0909c0b99f38474ca) the location of manifests in api/config/v1 & api/operator/v1 were changed after release-4.15.
Some PRs that reference to latest version of openshift/api fail due to the `files not found` at the path specified in the Dockerfile.

This PR bumps the api vendor to `@latest`, and updates the manifests path in the Dockerfile accordingly.

### CODE REVIEW

**Comment by adambkaplan** [comment]
/retitle OCPBUILD-9: Update openshift/api and vendored CRDs

**Comment by adambkaplan** [comment]
OpenShift repos now need JIRA references to merge :disappointed:

**Comment by adambkaplan** [comment]
/retest

The only failing test was the JenkinsPipeline build strategy, a thing which we've deprecated since OCP 4.10 (and I think earlier).

Removing this in OCP 4.16 or later release would not be the worst thing to consider...

**Review by adambkaplan — COMMENTED** [review]
/approve

Normally we try to separate the vendor bumps in a separate commit from non-vendor code changes.

File: pkg/cmd/operator/cmd.go [hunk 1]
```diff
@@ -3,6 +3,8 @@ package operator
 import (
 	"github.com/spf13/cobra"
 
+	_ "github.com/openshift/api/config/v1/zz_generated.crd-manifests"
+	_ "github.com/openshift/api/operator/v1/zz_generated.crd-manifests"
```
Comment (adambkaplan) [inline pkg/cmd/operator/cmd.go:7]: @sanchezl you might know the answer - should this go in https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/dependencymagnet/doc.go instead?

**Comment by adambkaplan** [comment]
Can the changes here also be included in #335?

**Comment by apoorvajagtap** [comment]
> Can the changes here also be included in #335?

@adambkaplan included these changes to #335. Also, I have imported [these packages](https://github.com/openshift/cluster-openshift-controller-manager-operator/pull/343#discussion_r1566356942) in [dependencymanagement/doc.go](https://github.com/openshift/cluster-openshift-controller-manager-operator/pull/335/files#diff-b05b59d94e1ed0e0fd67a8a12eef8a7f551d20c7597b62e9227bcb4eede88759). PTAL

**Comment by apoorvajagtap** [comment]
Closing this in favor of #335

**Pull request closed without merging**